WordPress/Classes/ViewRelated/Tools/HelpshiftPresenter.swift

```diff
@@ -0,0 +1,95 @@
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showConversation(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    /// Presents a Helpshift window displaying the FAQs index.
+    ///
+    /// - Parameters:
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - refreshUserDetails: If `true`, refresh user ID, display name, and email emailAddress
+    ///     from the WordPress.com REST API (if appropriate) before displaying the window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftFAQWindowFromViewController(viewController: UIViewController, refreshUserDetails: Bool, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showFAQs(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    private func updateHelpshiftUserDetailsWithUserID(userID: String, displayName: String, emailAddress: String) {
+        HelpshiftSupport.setUserIdentifier(userID)
+        HelpshiftCore.setName(displayName, andEmail: emailAddress)
+    }
+
+    private func prepareToDisplayHelpshiftWindow(refreshUserDetails: Bool, completion: () -> Void) {
+        NSUserDefaults.standardUserDefaults().setBool(true, forKey: UserDefaultsHelpshiftWasUsed)
+
+        PushNotificationsManager.sharedInstance.registerForRemoteNotifications()
+        InteractiveNotificationsManager.sharedInstance.registerForUserNotifications()
+
+        let context = ContextManager.sharedInstance().newDerivedContext()
+        let accountService = AccountService(managedObjectContext: context)
+
+        if let defaultAccount = accountService.defaultWordPressComAccount() {
+            if refreshUserDetails {
+                let remote = AccountServiceRemoteREST(wordPressComRestApi: defaultAccount.wordPressComRestApi)
+                remote.getDetailsForAccount(defaultAccount, success: { remoteUser in
+                    self.updateHelpshiftUserDetailsWithUserID(remoteUser.userID.stringValue,
+                                                              displayName: remoteUser.displayName,
+                                                              emailAddress: remoteUser.email)
+                    completion()
+                    }, failure: { _ in
+                        completion()
+                })
+
+                return
+            } else {
+                updateHelpshiftUserDetailsWithUserID(defaultAccount.userID.stringValue,
+                                                     displayName: defaultAccount.displayName,
+                                                     emailAddress: defaultAccount.email)
+            }
+        }
+
+        completion()
```

Comment:
The flow here isn't very clear. I thought you might be calling `completion` twice until I noticed the `return`.

How about this instead?

```swift
guard let defaultAccount = accountService.defaultWordPressComAccount() else {
    completion()
    return
}

updateHelpshiftUserDetailsWithUserID(defaultAccount.userID.stringValue,
                                     displayName: defaultAccount.displayName,
                                     emailAddress: defaultAccount.email)

guard refreshUserDetails else {
    completion()
    return
}

let remote = AccountServiceRemoteREST(wordPressComRestApi: defaultAccount.wordPressComRestApi)
remote.getDetailsForAccount(defaultAccount, success: { remoteUser in
    self.updateHelpshiftUserDetailsWithUserID(remoteUser.userID.stringValue,
        displayName: remoteUser.displayName,
        emailAddress: remoteUser.email)
    completion()
    }, failure: { _ in
        completion()
})
```